crates/node-builder/src/components/traits.rs

```diff
@@ -142,3 +155,37 @@ where
         self(ctx)
     }
 }
+
+/// Trait for launching the node.
+pub trait LaunchNode<DB, Types, Components>
+where
+    DB: Database + DatabaseMetrics + DatabaseMetadata + Clone + Unpin + 'static,
+    Types: Node<RethFullAdapter<DB, Types>>,
+    Types::PoolBuilder: PoolBuilder<RethFullAdapter<DB, Types>>,
+    Types::NetworkBuilder: crate::components::NetworkBuilder<
+        RethFullAdapter<DB, Types>,
+        <Types::PoolBuilder as PoolBuilder<RethFullAdapter<DB, Types>>>::Pool,
+    >,
+    Types::PayloadBuilder: crate::components::PayloadServiceBuilder<
+        RethFullAdapter<DB, Types>,
+        <Types::PoolBuilder as PoolBuilder<RethFullAdapter<DB, Types>>>::Pool,
+    >,
+    Components: NodeComponentsBuilder<RethFullAdapter<DB, Types>>,
+{
+    /// Launches the node and returns a handle to it.
+    ///
+    /// Returns a [NodeHandle] that can be used to interact with the node.
+    fn launch(
+        self,
+        config: NodeConfig,
+        state: RethFullBuilderState<DB, Types, Components>,
+        database: DB,
+        executor: TaskExecutor,
+        data_dir: ChainPath<DataDirPath>,
+        reth_config: reth_config::Config,
```

Comment:
can we put this on a container type, this way we can make changes to this more easily